worlds/yoshisisland/Rules.py

```diff
@@ -0,0 +1,603 @@
+    set_rule(world.multiworld.get_location("Tap-Tap The Red Nose's Fort: Stars", player), lambda state: logic.has_midring(state) and state.has_all({'Spring Ball', 'Large Spring Ball', 'Egg Plant', 'Key'}, player))
+
+    set_rule(world.multiworld.get_location("The Very Loooooong Cave: Red Coins", player), lambda state: state.has_all({'Chomp Rock'}, player) and (state.has('Egg Capacity Upgrade', player, 3) or logic.combat_item(state)))
+    set_rule(world.multiworld.get_location("The Very Loooooong Cave: Flowers", player), lambda state: state.has_all({'Chomp Rock'}, player) and (state.has('Egg Capacity Upgrade', player, 3) or logic.combat_item(state)))
+    set_rule(world.multiworld.get_location("The Very Loooooong Cave: Stars", player), lambda state: state.has_all({'Chomp Rock'}, player) and (state.has('Egg Capacity Upgrade', player, 3) or logic.combat_item(state)) and logic.has_midring(state))
+
+    set_rule(world.multiworld.get_location("The Deep, Underground Maze: Red Coins", player), lambda state: state.has_all({'Chomp Rock', 'Key', 'Large Spring Ball'}, player))
+    set_rule(world.multiworld.get_location("The Deep, Underground Maze: Flowers", player), lambda state: state.has_all({'Chomp Rock', 'Key', 'Large Spring Ball'}, player))
+    set_rule(world.multiworld.get_location("The Deep, Underground Maze: Stars", player), lambda state: state.has_all({'Chomp Rock', 'Tulip', 'Key'}, player) or (logic.has_midring(state) and state.has_all({'Key', 'Chomp Rock', 'Large Spring Ball'}, player)))
+    set_rule(world.multiworld.get_location("The Deep, Underground Maze: Level Clear", player), lambda state: state.has_all({'Chomp Rock', 'Key', 'Large Spring Ball', 'Dashed Platform'}, player))
+
+    set_rule(world.multiworld.get_location("KEEP MOVING!!!!: Red Coins", player), lambda state: (state.has('Egg Capacity Upgrade', player, 1) or logic.combat_item(state)))
+    set_rule(world.multiworld.get_location("KEEP MOVING!!!!: Flowers", player), lambda state: state.has_all({'Egg Plant'}, player))
+    set_rule(world.multiworld.get_location("KEEP MOVING!!!!: Stars", player), lambda state: logic.has_midring(state))
+
+    set_rule(world.multiworld.get_location("King Bowser's Castle: Red Coins", player), lambda state: state.has_all({'Helicopter Morph', 'Egg Plant'}, player) and logic._68CollectibleRoute(state))
+    set_rule(world.multiworld.get_location("King Bowser's Castle: Flowers", player), lambda state: state.has_all({'Helicopter Morph', 'Egg Plant'}, player) and logic._68CollectibleRoute(state))
+    set_rule(world.multiworld.get_location("King Bowser's Castle: Stars", player), lambda state: state.has_all({'Helicopter Morph', 'Egg Plant'}, player) and logic._68Route(state))
+
+    set_easy_extra_rules(world, player, logic)
+
+def set_easy_extra_rules(world: "YIWorld", player: int, logic: YoshiLogic) -> None:

```

Comment:
Nice!

Reply:
What exactly is this comment referring to?

Reply:
The extra methods that I remember I asked for last time